src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary/Counter/Counter32.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+#nullable disable
+
+namespace System.Collections.Concurrent
+{
+    /// <summary>
+    /// Scalable 32bit counter that can be used from multiple threads.
+    /// </summary>
+    internal sealed class Counter32: CounterBase
+    {
+        private class Cell
+        {
+            [StructLayout(LayoutKind.Explicit, Size = CACHE_LINE * 2 - OBJ_HEADER_SIZE)]
+            public struct SpacedCounter
+            {
+                [FieldOffset(CACHE_LINE - OBJ_HEADER_SIZE)]
+                public int count;
+            }
+
+            public SpacedCounter counter;
+        }
+
+        // spaced out counters
+        private Cell[] cells;
+
+        // default counter
+        private int count;
+
+        // delayed estimated count
+        private int lastCount;
+
+        /// <summary>
+        /// Initializes a new instance of the <see
+        /// cref="Counter32"/>
+        /// </summary>
+        public Counter32()
+        {
+        }
+
+        /// <summary>
+        /// Returns the value of the counter at the time of the call.
+        /// </summary>
+        /// <remarks>
+        /// The value may miss in-progress updates if the counter is being concurrently modified.
+        /// </remarks>
+        public int Value
+        {
+            get
+            {
+                var count = this.count;
+                var cells = this.cells;
+
+                if (cells != null)
+                {
+                    for (int i = 0; i < cells.Length; i++)
+                    {
+                        var cell = cells[i];
+                        if (cell != null)
+                        {
+                            count += cell.counter.count;
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                return count;
+            }
+        }
+
+        /// <summary>
+        /// Returns the approximate value of the counter at the time of the call.
+        /// </summary>
+        /// <remarks>
+        /// EstimatedValue could be significantly cheaper to obtain, but may be slightly delayed.
```

Comment:
❔ Is stale a better word choice here? Concern is the reader could interpret "slightly delayed" to mean "slower to return".

```suggestion
        /// EstimatedValue could be significantly cheaper to obtain, but may be stale.
```